### PR TITLE
1144 conform readme and creating contributing document

### LINKED ISSUE #1144: Conform ReadMe and create Contributing Document
There should be created CONTRIBUTING.md file for development purpose. One thing that needs to be written down to this file is SPM configuration, especially headers symlinks and where to place them in file structure (and why).

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-962) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by maratal**
Closes #1144

### CODE REVIEW

**Review by lukasz-szyszkowski — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,6 +26,18 @@ from the command line and then do a clean rebuild in Xcode.
...
+
+In order to make available new public APIs to the Carthage users you must put the symlink of your new header file into the `Source/include/Ably` folder. Also symlinks for all new private headers of internal classes (that ones with "+Private.h" suffix) should go into `Source/PrivateHeaders/Ably` folder, as well as symlinks for other private headers used by the framework (from `Source/Private` folder). This is for compatibility with `SPM` project structure.
+
+You should generate symlinks from the command line - in Terminal `cd` to the symlinks folder and type:
+
+```
+ln -s HeaderName.h ../../../Source/HeaderName.h
+```
+
+Do not use symlinks generated by macOS Finder app.
```
Comment (lawrence-forooghian) [inline CONTRIBUTING.md:39]: The Finder app doesn't generate symlinks – are you thinking of [aliases](https://en.wikipedia.org/wiki/Alias_(Mac_OS))? As described on that page, they're not the same thing.

File: README.md [hunk 2]
```diff
@@ -681,63 +681,4 @@ You can also view the [community reported Github issues](https://github.com/ably
...
-    4. Validate that the CocoaPods build should succeed using `pod lib lint`
-* If any fixes are needed (e.g. the lint fails with warnings) then either commit them to `main` branch now if they are simple warning fixes or perhaps consider raising a new PR if they are complex or likely to need review.
-* Create a tag for this version number using `git tag x.x.x`
-* Push the tag using `git push origin x.x.x`
-* Release an update for CocoaPods using `pod trunk push Ably.podspec`. Details on this command, as well as instructions for adding other contributors as maintainers, are at [Getting setup with Trunk](https://guides.cocoapods.org/making/getting-setup-with-trunk.html) in the [CocoaPods Guides](https://guides.cocoapods.org/)
-* Add to [releases](https://github.com/ably/ably-cocoa/releases)
-    * refer to previous releases for release notes format
-    * attach the prebuilt framework file generated by Carthage to the release: `Ably.framework.zip`
-* Test the integration of the library in a Xcode project using Carthage and CocoaPods using the [installation guide](https://github.com/ably/ably-cocoa#installation-guide)
+See [Contributing.md](./Contributing.md) for details.
```
Comment (lawrence-forooghian) [inline README.md:684]:
```suggestion
See [CONTRIBUTING.md](./CONTRIBUTING.md) for details.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+In this repository the `main` branch contains the latest development version of the Ably SDK. All development (bug fixing, feature implementation, etc.) is done against the `main` branch, which you should branch from whenever you'd like to make modifications. Here's the steps to follow when contributing to this repository.
```
Comment (lawrence-forooghian) [inline CONTRIBUTING.md:1]: The headers aren't quite right in this document - needs an `h1` and the current `h3`s should be `h2`s.

File: CONTRIBUTING.md (hunk 1 above)
Comment (maratal) [reply CONTRIBUTING.md]: ee6e06dd

File: CONTRIBUTING.md (hunk 3 above)
Comment (maratal) [reply CONTRIBUTING.md:3]: c86eb036

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**